### ISSUE TITLE
XP bar color requires a refresh

### ISSUE THREAD

**Issue by foresthoffman** [TheMacLab/game-on#125]
The color of the XP bar in the admin bar doesn't update when a user receives damage. A refresh is required to have the XP bar display the correct color. This appears to be from [`v2.5.9`](https://github.com/TheMacLab/game-on/tree/v2.5.9).

**Comment by AnthoneyKalasho** [comment]
@foresthoffman Hey I'm new to the project, I'm still getting familiar with the source but I've been working on this issue because it seems simple enough

**Comment by foresthoffman** [comment]
Howdy @AnthoneyKalasho. If you can sort this out that would be very helpful. I'm swamped at the moment, but feel free to fork and then make a pull request when you're done.

Any changes you make should be done on top of [`v2.6.0`](https://github.com/TheMacLab/game-on/releases/v2.6.0), since [`v3.0.0`](https://github.com/TheMacLab/game-on/releases/v3.0.0) is still a pre-release.

I look forward to seeing what you've got. Thank you!

**Comment by AnthoneyKalasho** [comment]
@foresthoffman 
Kind of embarrassed that this is taking me so much longer than I expected it to... I figured out a solution in like two days, echoing the following lines inside `buy-ajax.php` at line 112.

    $color = barColor( $current_bonus_currency, $current_penalty );
    echo '<script type="type/javascript">';
    echo '$("go_admin_bar_progress_bar").css("background-color","'.$color.' !important");';
    echo '</script>';

While it successfully updates the XP bar color likes its supposed to, adding any scripts at this spot seems to hang the buy function and I can't figure out why. I'll keep working at it but if you have any idea to what it might be that'd really help.

**Comment by Charlesleonius** [comment]
If you're using Ajax then your php and JavaScript should be separate. Don't
echo out your script but instead have one that fires based on the result of
your Ajax call.

**Comment by foresthoffman** [comment]
@AnthoneyKalasho first off, you have absolutely no reason to feel embarrassed. You're jumping into a massive pile of spaghetti code that has been built up over 3 years. :) Give yourself a pat on the back for taking the plunge.

Second, @Charlesleonius is absolutely correct. Going forward, we want to utilize WordPress's script and style enqueuing hooks. That is why I created `go_enqueue.php` (see the file in v2.6.0, [here](https://github.com/TheMacLab/game-on/blob/v2.6.0/go_enque.php)). Where applicable, we want to adhere to this rule.

That said, I do not believe that you should be looking at `buy-ajax.php` (aside from using it for debugging). Trying to update the admin bar's styles via a JS block in that very specific case will only solve the issue for purchasing non-gifted store items that have penalties attached to them. We want the admin bar styles to be consistent across all the features in Game On.

I suggest that you look at the [`go_update_admin_bar()`](https://github.com/TheMacLab/game-on/blob/v2.6.0/go_pnc.php#L289) function in `go_pnc.php`. That is the function that handles the output of the JS that updates the XP bar in the admin bar. The [`barColor()`](https://github.com/TheMacLab/game-on/blob/v2.6.0/go_pnc.php#L478) function which actually returns the hexcode for the XP bar colors. Also check out the [`go_return_multiplier()`](https://github.com/TheMacLab/game-on/blob/v2.6.0/go_pnc.php#L502) function which runs calculations on currency values using (and can return) what we call the "Super Modifier".

As far as debugging goes, I would output (to your php_error.log) the arguments for each of those functions. That will allow you to analyze what is going in and out of each of those functions. You may find something funky.

At a glance, the `barColor()` function doesn't appear to take the Super Modifier into account. That could be a potential source of unexpected behavior in the XP bar as well.

> **Where applicable**, we want to adhere to this rule.

That's where this comes into play...I know that `go_update_admin_bar()` doesn't follow the standard, but currently a small graphical glitch doesn't justify refactoring the whole thing.

===

I apologize if that is hard to follow, I'm a bit scatter-brained currently, so the comment may reflect that. Let me know and i'll try to clarify if I can.

**Comment by Charlesleonius** [comment]
I may be able to provide some further input later today as I originally implemented the color changing system and had it working in previous versions.

**Comment by AnthoneyKalasho** [comment]
I figured out that commenting out line 51 of buy_the_item.js fixes the issue but I'm not sure what the purpose of this line was to begin with. Hopefully this means I'm a step closer to a solution...

**Comment by foresthoffman** [comment]
If I'm looking at the correct line (51):
```js
jQuery( '#go_admin_bar_progress_bar' ).css( "background-color", color );
```
This is supposed to update the admin bar's background color.

On line 13, you can see the corresponding declaration of `color`:
```js
var color = jQuery( '#go_admin_bar_progress_bar' ).css( 'background-color' );
```
At a glance, it doesn't seem useful. It may be a band-aid fix to deal with some unusual styling behavior. I don't quite remember if that's the case, without diving into it. @Charlesleonius?
